subsystems/visionsubsystem.py

```diff
@@ -1,76 +1,88 @@
+    def __init__(self) -> None:
         Subsystem.__init__(self)
         self.setName(__class__.__name__)
-        self.drive = drive
-        self.estimatedPosition = Pose2d()
+        # self.estimatedPosition = Pose2d()
 
-        self.camera = PhotonCamera(constants.kPhotonvisionCameraName)
+        # self.camera = PhotonCamera(constants.kPhotonvisionCameraName)
 
+        self.cameras = [
+            PhotonCamera(camera) for camera in constants.kPhotonvisionCameraArray
+        ]
+        self.poseList = deque([])
         # if RobotBase.isSimulation():
         #     inst = NetworkTableInstance.getDefault()
         #     inst.stopServer()
         #     inst.setServer("localhost")
         #     inst.startClient4("Robot Sim")
 
     def periodic(self) -> None:
-        self.estimatedPosition = self.drive.getPose()
-        self.updateAdvantagescopePose()
-
-        photonResult = self.camera.getLatestResult()
-        hasTargets = len(photonResult.getTargets()) > 0
-        multitagresult = photonResult.multiTagResult
+        # self.estimatedPosition = self.drive.getPose()
+        # self.updateAdvantagescopePose()
+        for camera in self.cameras:
+            photonResult = camera.getLatestResult()
+            hasTargets = len(photonResult.getTargets()) > 0
+            multitagresult = photonResult.multiTagResult
+            bestRelativeTransform = multitagresult.estimatedPose.best
 
-        bestRelativeTransform = multitagresult.estimatedPose.best
-
-        if not multitagresult.estimatedPose.isPresent:
-            ambiguity = 10
-            for result in photonResult.targets:
-                if result.poseAmbiguity < ambiguity:
-                    bestRelativeTransform = (
-                        Transform3d(
-                            Pose3d(), constants.kApriltagPositionDict[result.fiducialId]
+            if not multitagresult.estimatedPose.isPresent:
+                ambiguity = 10
+                for result in photonResult.targets:
+                    if result.poseAmbiguity < ambiguity:
+                        bestRelativeTransform = (
+                            Transform3d(
+                                Pose3d(),
+                                constants.kApriltagPositionDict[result.fiducialId],
+                            )
+                            + result.bestCameraToTarget.inverse()
                         )
-                        + result.bestCameraToTarget.inverse()
-                    )
-                    ambiguity = result.poseAmbiguity
+                        ambiguity = result.poseAmbiguity
+
+            self.updateAdvantagescopePose(Pose3d() + bestRelativeTransform, camera)
 
-        botPose = (
-            Pose3d()
-            + bestRelativeTransform
-            + constants.kLimelightRelativeToRobotTransform.inverse()
-        )
+            cameraTransforms = {
+                "frontLeft": constants.kRobotToFrontLeftCameraTransform,
```

Comment:
this, as well as the keys, and the cameras should get grouped into a larger struct like EstimatedPose
something like the following
```py
class VisionCamera:
    transform: Transform3d
    camera: PhotonCamera
    poseKey: str
```